teams/vsp/teams/tools/frontend/cypress-migration-guide.md

```diff
@@ -0,0 +1,242 @@
+
+```javascript
+cy.get('[name="root_attachments_0_attachmentName"]')
+  .upload('src/platform/testing/example-upload.png', 'image/jpg')
+```
+
+### Accessibility <a name="accessibility"></a>
+Just like most e2e tools, cypress has its own [axe-core plugin](https://github.com/avanslaars/cypress-axe) to test accessibility.
+
+To add axe checks to your tests use the custom `axeCheck()` command based off the `checkA11y` [command](https://github.com/avanslaars/cypress-axe#cychecka11y).
+
+#### Nightwatch:
+```javascript
+client.axeCheck('.main')
+```
+
+#### Cypress:
+```javascript
+cy.axeCheck();
+```
+
+### Assertions <a name="assertions"></a>
+
+Cypress uses `chai` to [handle assertions](https://docs.cypress.io/guides/references/assertions.html#BDD-Assertions). Currently in the `nightwatch-assertions` folder, we have 4 custom assertions, which can all be replaced:
+
+#### hasFocusableCount / hasTabbableCount
+Checks the count of focusable and tabbable elements.
+
+These can be converted by making [custom chai assertions](https://docs.cypress.io/guides/references/assertions.html#Adding-New-Assertions).
+
+#### isActiveElement
+Checks if the given element is focused on the page.
+
+```javascript
+// Bdd assertion
+cy.get('input').should('have.focus')
+```
+#### isDisabledElement 
+Checks if the given element is disabled on the page.
+
+```javascript
+// Bdd assertion
+cy.get('button').should('be.disabled')
+```
+
+## Running Tests <a name="running-tests"></a>
+
+As of now, cypress supports Chrome, Edge, Firefox(beta) and a few [others](https://docs.cypress.io/guides/guides/launching-browsers.html#Browsers). You can run tests in headless mode or via the test runner.
+
+### Headless mode <a name="headless-mode"></a>
+To run tests headless, run `yarn test:cypress:run`
+
+### Test Runner <a name="test-runner"></a>
+To run tests using the cypress [test runner](https://docs.cypress.io/guides/core-concepts/test-runner.html#Overview), run `yarn test:cypress:open`.
+
+With the Test Runner, you are able to use cypress's `Open Selector Playground'. This allows you to click on elements in the DOM and copy the selector for that item for your test. This means there's usually no need to inspect an element manually.
+
+The test runner provides the ability to pause tests, and [time travel](https://docs.cypress.io/guides/getting-started/writing-your-first-test.html#Time-travel), which allows you to see snapshots of your tests.
+
+You can add any of the [run options](https://docs.cypress.io/guides/guides/command-line.html#Commands) to the commands above.
+
+## Things to Note <a name="things-to-note"></a>
+
+### Automatic Waiting
+Cypress automatically waits for commands to execute before moving on to the next. This eliminates needing to use the timeout consts in `platform/testing/e2e/timeouts.js`
```

Comment:
No example off the top of my head, but I think it might be worth pointing out that Cypress queues up its commands instead of runniing them synchronously. Clearing that up for people might help them avoid a painful ordeal with trying to combine Cypress commands with synchronous code.

Reply:
[This might work.](https://docs.cypress.io/guides/references/best-practices.html#Assigning-Return-Values)